docs: document installation process (#81)
officelifehq/docs: docs/documentation/installation.md

  * PDO PHP Extension
  * Tokenizer PHP Extension
  * XML PHP Extension
* MySQL, Mariadb or PostgreSQL
* Apache or Nginx

### Deployment

- We recommend [Fortrabbit](https://www.fortrabbit.com/) and [Platform.sh](https://platform.sh/) as PaaS (Platform As A Service) hosts, or [Forge](https://forge.laravel.com) and [Ploi](https://ploi.io/) as IaaS platforms provisioner, to deploy an OfficeLife instance.

- You can also use our official [Docker image](https://github.com/officelifehq/docker/pkgs/container/officelife-dev) if you prefer.

> - You can also download a bundle asset from our [releases](https://github.com/officelifehq/officelife/releases) containing all dependencies at once. The assets also contains gnupg signature files, signed off with the key `0x213F85B7B1B8F93224FA85B138C2BAD715449E85`. You will find the public key on several [keyserver](http://keyserver.ubuntu.com/pks/lookup?search=0x213F85B7B1B8F93224FA85B138C2BAD715449E85&fingerprint=on&op=index), or [here](pubkey.asc).

Reviewer: ```suggestion
- You can also download a bundle asset from our [releases](https://github.com/officelifehq/officelife/releases) containing all dependencies at once. The assets also contains gnupg signature files, signed off with the key `0x213F85B7B1B8F93224FA85B138C2BAD715449E85`. You will find the public key on several [keyservers](http://keyserver.ubuntu.com/pks/lookup?search=0x213F85B7B1B8F93224FA85B138C2BAD715449E85&fingerprint=on&op=index), or [here](pubkey.asc).
```

Reviewer: @asbiin also, are you sure `http://keyserver.ubuntu.com/pksz` is not `https`?

Author: You're right, we can use https here!

### Setup

You should configure your web server's document / web root to be the `public` directory. The `index.php` in this directory serves as the front controller for all HTTP requests entering your application.

Once installed, run this command to setup everything:
```
php artisan setup
```

### Directory permissions

You may need to configure some permissions. Directories within the `storage` and the `bootstrap/cache` directories should be writable by your web server or OfficeLife will not run.

## Configuration

All configuration are set through the `.env` file or within environment variables.

> The `.env.example` file contains a list of common configuration you might want to use. These are the principal ones:

Reviewer: ```suggestion
The `.env.example` file contains a list of common configuration you might want to use. These are the main ones:
```

### Global config

- `APP_ENV`: set to `production` for a production installation
- `APP_KEY`: this is provisionned by the `php artisan key:generate` command, or you can manually generate it using this command: `echo -n 'base64:'; openssl rand -base64 32`

> - `APP_DEBUG`: this should be set to `false` for a production installation

Reviewer: ```suggestion
- `APP_DEBUG`: this should be set to `false` for a production installation. If set to `true`, this will display all debug traces when an error occurs, so you should never, ever set it to `true` for production use.
```

#### LinkedIn

On LinkedIn developers portal, create a new app. In Auth setting, setup OAuth 2.0 settings using `https://yourapp.com/auth/linkedin/callback` as Authorized redicret URL.

- `LINKEDIN_CLIENT_ID`: the client ID
- `LINKEDIN_CLIENT_SECRET`: the client Secret

#### Saml2

Saml2 is a generic provider standard. The callback UI will be `https://yourapp.com/auth/saml2/callback`.
You can add a name to your Saml2 application by setting `SAML2_NAME`.

Once you have setup you saml2 application, you can either
- Set the `SAML2_METADATA` to the metadata url
- Or set `SAML2_ACS`, `SAML2_ENTITY_ID`, `SAML2_CERTIFICATE` with the right values

#### Slack

On slack API, create a new app, and setup OAuth & Permissions, using `https://yourapp.com/auth/slack/callback` as Redirect URLS, and adding `identity` User Token Scopes.

- `SLACK_CLIENT_ID`: the client ID
- `SLACK_CLIENT_SECRET`: the client Secret

#### Twitter

On twitter developer portal, create a new app or project. Activate the 3-legged OAuth authentication, and set `https://yourapp.com/auth/twitter/callback` as Callback url.

`TWITTER_CLIENT_ID`
`TWITTER_CLIENT_SECRET`

## Cron and queue

> OfficeLife is a Laravel-based application, and requires a cron to run, and eventually a queue worker.

Reviewer: ```suggestion
OfficeLife is a Laravel-based application, and requires a cron to run, and possibly a queue worker if you can.
```

### Cron
The cron is a job that run at regular intervals, ideally every minutes.
Setup a cron that run this command:
```
php artisan schedule:run
```
See [Laravel documentation](https://laravel.com/docs/8.x/scheduling#running-the-scheduler)

### Queue

> Setup a queue is a very good way to improve performances on OfficeLife.

Reviewer: ```suggestion
Setting up a queue is a very good way to improve performances on OfficeLife.
```

* Apache or Nginx

### Deployment

- We recommend [Fortrabbit](https://www.fortrabbit.com/) and [Platform.sh](https://platform.sh/) as <abbr title="Platform As A Service">PaaS</abbr> hosts, or [Forge](https://forge.laravel.com) and [Ploi](https://ploi.io/) as <abbr title="Infrastructure As A Service">IaaS</abbr> platforms provisioner, to deploy an OfficeLife instance.

- You can also use our official [Docker image](https://github.com/officelifehq/docker/pkgs/container/officelife-dev) if you prefer.

- You can also download a bundle asset from our [releases](https://github.com/officelifehq/officelife/releases) containing all dependencies and build files at once. The assets also contains gnupg signature files, signed off with the key `0x213F85B7B1B8F93224FA85B138C2BAD715449E85`. You will find the public key on several [keyservers](https://keyserver.ubuntu.com/pks/lookup?search=0x213F85B7B1B8F93224FA85B138C2BAD715449E85&fingerprint=on&op=index), or [here](/pubkey.asc).

#### Build

If you choose to deploy from the sources, you will have to build the application.

_**Requirements**_
- composer
- yarn

_**Steps**_
1. `composer install`
2. `yarn install`
3. `yarn run production`

> Then the `vendor`, and the files created in `public` folder must be deployed within the application (the `node_modules` folder can be removed).

Reviewer: ```suggestion
Then the `vendor` directory as well as the files created in `public` folder must be deployed within the application (the `node_modules` folder can be removed).
```

@asbiin deployed how? where?

### Mailing
Define your mailing system to use:

- `MAIL_MAILER`: the mailer config to use: `smtp`, `sendmail`, `mailgun`, `ses`, or `postmark`
- Set these variable according to your configuration:
`MAIL_HOST`,
`MAIL_PORT`,
`MAIL_USERNAME`,
`MAIL_PASSWORD`,
`MAIL_ENCRYPTION`
- `MAIL_FROM_ADDRESS` and `MAIL_FROM_NAME` for outbound mail information

### Services

- `LOCATION_IQ_API_KEY`: API key for [LocationIQ](https://locationiq.com/) geolocation service. LocationIQ offers 10,000 free requests per day.
- `MAPBOX_API_KEY` and `MAPBOX_USERNAME`: API key for maps displays using [Mapbox](https://mapbox.com/).
- `CURRENCY_LAYER_API_KEY` and `CURRENCY_LAYER_PLAN` (`free` or `premium`): API key for getting currency exchange rates, using [CurrencyLayer](https://currencylayer.com/).
- `UPLOADCARE_PUBLIC_KEY` and `UPLOADCARE_PRIVATE_KEY`: API key for uploading files in [Uploadcare](https://uploadcare.com).

### External login providers

You can add external login providers to OfficeLife.

- `LOGIN_PROVIDERS`: a coma-separated list of enabled login providers. Available providers are: `azure`, `github`, `google`, `linkedin`, `monica`, `saml2`, `slack`, `twitter`.

> An enabled login provider will be showned on the login page. The order of the list is the order these providers will be displayed.

Reviewer: ```suggestion
An enabled login provider will be shown on the login page. The order of the list is the order these providers will be displayed.
```
